_posts/2020-07-31-fnn-vae-for-noisy-timeseries/fnn_vae_for_noisy_timeseries.Rmd

```diff
@@ -0,0 +1,506 @@
+knitr::include_graphics("images/noise_1_nofnn.png")
+```
+
+What happens when we start to add noise?
+
+#### Substantial noise
+
+Between noise levels 1.5 and 2, something changed, or became noticeable from visual inspection. Let's jump directly to the
+highest-used level though: 2.5.
+
+Here first are predictions obtained from the unregularized models.
+
+```{r, eval=TRUE, echo=FALSE, layout="l-page", fig.cap = "Roessler series with added Gaussian noise of standard deviation 2.5. Grey: actual (noisy) test data. Green: underlying Roessler system. Orange: Predictions from unregularized LSTM. Dark blue: Predictions from unregularized VAE."}
+knitr::include_graphics("images/noise_2.5_nofnn.png")
+```
+
+Both LSTM and VAE get "distracted" a bit too much by the noise, the latter to an even higher degree. This leads to cases
+where predictions strongly "overshoot" the underlying non-noisy rhythm. This is not surprising, of course: They were *trained*
+on the noisy version; predict fluctuations is what they learned.
+
+Do we see the same with the FNN models?
+
+```{r, eval=TRUE, echo=FALSE, layout="l-page", fig.cap = "Roessler series with added Gaussian noise of standard deviation 2.5. Grey: actual (noisy) test data. Green: underlying Roessler system. Orange: Predictions from FNN-LSTM. Dark blue: Predictions from FNN-VAE."}
+knitr::include_graphics("images/noise_2.5.png")
+```
+
+Interestingly, we see a much better fit to the underlying Roessler system now! Especially the VAE model, FNN-VAE, surprises
+with a whole new smoothness of predictions; but FNN-LSTM turns up much smoother forecasts as well.
+
+"Smooth, fitting the system..." -- by now you may be wondering, when are we going to come up with more quantitative
+assertions? If quantitative implies "mean squared error" (MSE), and if MSE is taken to be some divergence between forecasts
+and the true target from the test set, the answer is that this MSE doesn't differ much between any of the four architectures.
+Put differently, it is mostly a function of noise level.
+
+However, we could argue that what we're really interested in is how well a model forecasts the underlying process. And there,
+we see differences.
+
+In the following plot, we contrast MSEs obtained for the four model types (grey: VAE; orange: LSTM; dark blue: FNN-VAE; green:
+FNN-LSTM). The rows reflect noise levels (1, 1.5, 2, 2.5); the columns represent MSE in relation to the noisy("real") target
+(left) on the one hand, and in relation to the underlying system on the other (right). For better visibility of the effect,
+*MSEs have been normalized as fractions of the maximum MSE in a category*.
+
+So, if we want to predict *signal plus noise* (left), it doesn't really matter whether we use FNN or not. But if we want to
```

Comment:
I wouldn't say it doesn't matter as from the plot it seems that FNN regularised versions are consistently slightly better for higher noise values.

Reply:
you're right! I really wanted to be as cautious as possible ... But if you say so, i'll rephrase that a bit :-)